extensions/jira-time-tracking/src/index.tsx

```diff
@@ -0,0 +1,162 @@
+  const [minutes, setMinutes] = React.useState("0");
+  const [seconds, setSeconds] = React.useState("0");
+  const [description, setDescription] = React.useState("");
+  const [selectedProject, setSelectedProject] = React.useState<string>();
+  const [startedAt, setStartedAt] = React.useState<Date>(new Date());
+  const [loading, setLoading] = React.useState(true);
+
+  async function handleSubmit() {
+    const totalTimeWorked = toSeconds(Number(seconds), Number(minutes), Number(hours));
+
+    if (!selectedIssue) {
+      showToast(ToastStyle.Failure, "Error logging time: issue not found");
+      return;
+    }
+    if (!totalTimeWorked) {
+      showToast(ToastStyle.Failure, "Error logging time: no time entered.");
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      await postTimeLog(totalTimeWorked, selectedIssue.key, description, startedAt);
+      const successMessage = createTimeLogSuccessMessage(selectedIssue.key, hours, minutes, seconds);
+      showToast(ToastStyle.Success, successMessage);
+      cleanUp();
+    } catch (e) {
+      const message = e instanceof Error ? e.message : "Error Logging Time";
+      showToast(ToastStyle.Failure, message);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  // fetch projects on mount
+  React.useEffect(() => {
+    const fetchProjects = async () => {
+      try {
+        const projects = await getProjects();
+        if (projects.length) {
+          setProjects(projects);
+        }
+      } catch (e) {
+        const message = e instanceof Error ? e.message : "Error Logging Time";
+        showToast(ToastStyle.Failure, message);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchProjects();
+  }, []);
+
+  // fetch issues after project is selected
+  React.useEffect(() => {
+    if (selectedProject) {
+      const fetchIssues = async () => {
+        const issues = await getIssues(selectedProject);
+        if (issues) {
+          setIssues(issues);
+          setSelectedIssue(issues.at(-1));
+        }
+      };
+      fetchIssues();
+    }
+  }, [selectedProject]);
+
+  const handleSelectIssue = (key: string) => {
+    const issue = issues.find((issue) => issue.key === key);
+    if (issue) setSelectedIssue(issue);
+  };
+
+  const cleanUp = () => {
+    setDescription("");
+    setHours("0");
+    setMinutes("0");
+    setSeconds("0");
+  };
+
+  const emptyMessage = `
+# No projects found
+
+No Jira projects were found using your credentials.
+
+This could happen because:
+
+- The provided jira domain has no associated projects.
+- The email credential provided is not authorized to access any projects on the provided jira domain.
+- The email credential provided is incorrect.
+- The API token credential provided is incorrect.
+
+Please check your permissions, jira account or credentials and try again.
+  `;
+
+  if (!projects && !loading) {
+    return <Detail markdown={emptyMessage} />;
+  }
```

Comment:
One thing that could be nice to see if we can address would be removing the flicker before showing this. Do we need `!loading`?

Reply:
Nice idea @pernielsentikaer! Unfortunately I can't think of a good solution to this at the minute. If we remove the loading prop we will have the converse problem: if the user does successfully fetch some projects, the <Detail/> will flicker before the <Form/> component. Since the <Form/> component is the primary flow (and the <Detail/> just a fallback in case there are no projects for whatever reason), I figured the flicker would be better on that side of things. Also, since we must return some component (AFAIK) using the raycast API, we can't return null or undefined while the data is fetching. What do you reckon?